exp-player/addon/components/exp-lookit-dialogue-page/component.js

```diff
@@ -0,0 +1,797 @@
+            "autoProceed": false,
+            "isChoiceFrame": true,
+            "parentTextBlock": {
+                "title": "Parents:",
+                "text": "click on the character your child selects.",
+                "emph": true
+            },
+            "images": [
+                {
+                    "id": "speaker1",
+                    "src": "order1_test1_speaker1.png",
+                    "left": "20",
+                    "bottom": "2",
+                    "height": "60"
+                },
+                {
+                    "id": "speaker2",
+                    "src": "order1_test1_speaker2.png",
+                    "left": "60",
+                    "bottom": "2",
+                    "height": "60"
+                }
+            ],
+            "audioSources": [
+                {
+                    "audioId": "firstAudio",
+                    "sources": [
+                        {
+                            "stub": "polcon_example_5q1"
+                        }
+                    ]
+                }
+            ]
+        }
+ }
+
+ * ```
+ * @class ExpLookitDialoguePage
+ * @extends ExpFrameBaseUnsafe
+ * @uses FullScreen
+ * @uses VideoRecord
+ */
+
+export default ExpFrameBaseUnsafeComponent.extend(FullScreen, VideoRecord,  {
+    // In the Lookit use case, the frame BEFORE the one that goes fullscreen
+    // must use "unsafe" saves (in order for the fullscreen event to register as
+    // being user-initiated and not from a promise handler) #LEI-369.
+    // exp-alternation frames are expected to be repeated, so they need to be
+    // unsafe.
+    type: 'exp-lookit-dialogue-page',
+    layout: layout,
+    displayFullscreen: true, // force fullscreen for all uses of this component
+    fullScreenElementId: 'experiment-player',
+    fsButtonID: 'fsButton',
+    videoRecorder: Ember.inject.service(),
+    recorder: null,
+    hasCamAccess: Ember.computed.alias('recorder.hasCamAccess'),
+    videoUploadConnected: Ember.computed.alias('recorder.connected'),
+
+    // Track state of experiment
+    completedAudio: false, // for main narration audio
+    imageAudioCompleted: new Set(),
+    currentlyHighlighted: null, // id for image currently selected
+
+    currentAudioIndex: -1, // during initial sequential audio, holds an index into audioSources
+
+    // Can the user click the 'next' button yet? Require all 'main' audio to
+    // have played. For a choice frame, require that one of the images is
+    // selected; for other frames, require that any required image-audio has
+    // completed.
+    readyToProceed: Ember.computed('completedAudio', 'imageAudioCompleted', 'currentlyHighlighted',
+        function() {
+            var okayToProceed = this.get('completedAudio');
+
+            if (this.get('isChoiceFrame') && !(this.get('currentlyHighlighted'))) {
+                okayToProceed = false;
+            } else {
+                var whichAudioCompleted = this.imageAudioCompleted;
```

Comment:
Pedantic note: we generally prefer to manipulate properties of a component with `.get` / `.set`. 

Partly for consistency, partly because (for setters), use of the right method is one of several hints to help ember ensure that dependent computed properties know to update.